packages/ai-chat/src/common/command-chat-agents.ts

```diff
@@ -0,0 +1,412 @@
+
+workbench.action.selectIconTheme
+
+## Invalid Response Example 3
+
+The following example is invalid because it returns a message with the command id. We need JSON objects based on the above rules.
+Do not respond like this in any case! We need a command of \`"type": "theia-command"\`.
+
+The expected response would be:
+\`\`\`json
+{
+    "type": "theia-command",
+    "commandId": "core.close.right.tabs"
+}
+\`\`\`
+
+### The Example
+
+I found this command that might help you: core.close.right.tabs
+
+## Invalid Response Example 4
+
+The following example is invalid because it has an explanation string before the JSON. 
+We only want the JSON!
+
+### The Example
+
+You can toggle high contrast mode with this command:
+
+\`\`\`json
+{
+    "type": "theia-command",
+    "commandId": "editor.action.toggleHighContrast"
+}
+\`\`\`
+
+## Invalid Response Example 5
+
+The following example is invalid because it explains that no command was found. 
+We want a response of \`"type": "no-command"\` and have the message there.
+
+### The Example
+
+There is no specific command available to "open the windows" in the provided Theia command list.
+
+## Invalid Response Example 6
+
+In this example we were using the following theia id command list:
+
+Begin List:
+container--theia-open-editors-widget: Hello
+foo:toggle-visibility-explorer-view-container--files: Label 1
+foo:toggle-visibility-explorer-view-container--plugin-view: Label 2
+End List
+
+The problem is that workbench.action.toggleHighContrast is not in this list. 
+theia-command types may only use commandIds from this list. 
+This should have been of \`"type": "no-command"\`.
+
+### The Example
+
+\`\`\`json
+{
+    "type": "theia-command",
+    "commandId": "workbench.action.toggleHighContrast"
+}
+\`\`\`
+
+`;
+}
+
+interface ParsedCommand {
+    type: 'theia-command' | 'custom-handler' | 'no-command'
+    commandId: string;
+    arguments?: string[];
+    message?: string;
+}
+
+@injectable()
+export class CommandChatAgent implements ChatAgent {
+
+    @inject(PromptService)
+    protected promptService: PromptService;
+
+    @inject(CommandRegistry)
+    protected readonly commandRegistry: CommandRegistry;
+
+    @inject(MessageService)
+    private readonly messageService: MessageService;
+
+    @inject(CommunicationRecordingService)
+    protected recordingService: CommunicationRecordingService;
+
+    @inject(LanguageModelRegistry)
+    protected languageModelRegistry: LanguageModelRegistry;
+
+    id: string = 'CommandChatAgent';
+    name: string = 'Command Chat Agent';
+    description: string = 'The default chat agent provided by Theia responsible for providing commands.';
+    variables: string[] = [];
+    promptTemplates: PromptTemplate[] = [new CommandChatAgentSystemPromptTemplate()];
+    languageModelRequirements: Omit<LanguageModelSelector, 'agent'>[] = [{
+        purpose: 'command',
+        identifier: 'openai/gpt-4o',
+    }];
+    locations: ChatAgentLocation[] = [];
```

Comment:
```suggestion
    locations: ChatAgentLocation[] = ChatAgentLocation.ALL;
```
Same as above the Vscode parser for agent selection will fail/ fallback to default if the locations array is empty